libcudacxx/include/cuda/std/__expected/expected_base.h

```diff
@@ -236,15 +237,16 @@ struct __expected_destruct<_Tp, _Err, false, false>
       , __has_val_(false)
   {}
 
+  _CCCL_EXEC_CHECK_DISABLE
   _LIBCUDACXX_INLINE_VISIBILITY _CCCL_CONSTEXPR_CXX20 ~__expected_destruct()
   {
     if (__has_val_)
     {
-      _CUDA_VSTD::__destroy_at(_CUDA_VSTD::addressof(__union_.__val_));
```

Comment:
Question: there is no advantage of `destroy_at` over calling the dtor explicitely? Both work during constant evaluation, if the dtor is `constexpr. `destroy_at` only has a convenience advantage because we do not need to spell the destroyed type's dtor, right?

Reply:
Right, but it has the disadvantage of an additional function instantiation and that it will only work with `std::destroy_at` during compile time as that is the standard blessed function